src/cleanfigure.m

```diff
@@ -743,89 +750,200 @@ function movePointsCloser(meta, handle)
+  end
+
+  % Extract the data from the current line handle.
+  xData = get(handle, 'XData');
+  yData = get(handle, 'YData');
+
+  % Update the data indicated by id_update
+  xData(id_replace) = dataReplace(:,1);
+  yData(id_replace) = dataReplace(:,2);
+
+  % Set the new (masked) data.
+  set(handle, 'XData', xData);
+  set(handle, 'YData', yData);
+end
+% =========================================================================
+function insertData(handle, id_insert, dataInsert)
+  % Inserts the elements of the cell array dataInsert at position id_insert
+
+  % Only do something if id_insert is not empty
+  if isempty(id_insert)
+      return
+  end
+
+  % Extract the data from the current line handle.
+  xData = get(handle, 'XData');
+  yData = get(handle, 'YData');
+
+  length_segments = [id_insert(1);
+                     diff(id_insert);
+                     length(xData)-id_insert(end)];
+
+  % Put the data into one matrix
+  data     = [xData(:), yData(:)];
+
+  % Cut the data into segments
+  dataCell = mat2cell(data, length_segments, 2);
+
+  % Merge the cell arrays
+  dataCell = [dataCell';
+              dataInsert'];
+
+  % Merge the cells back together
+  data     = cat(1, dataCell{:});
+
+  % Set the new (masked) data.
+  set(handle, 'XData', data(:,1));
+  set(handle, 'YData', data(:,2));
+end
+% =========================================================================
+function removeData(meta, handle, id_remove)
+  % Removes the data at position id_remove
+
+  % Only do something if id_remove is not empty
+  if isempty(id_remove)
+      return
+  end
   
-      % Rotational matrix with sign flip. It transforms a given vector [a,b] by 
-      % Rotate * [a,b] = [-b,a] as required for calculation of invA  
-      Rotate = [0, -1; 1, 0];   
+  % Extract the data from the current line handle.
+  xData = get(handle, 'XData');
+  yData = get(handle, 'YData');
+  if isAxis3D(meta.gca)
+  	zData = get(handle, 'ZData');
+  end
 
+  % Remove the data indicated by id_remove
+  xData(id_remove) = [];
+  yData(id_remove) = [];
+  if isAxis3D(meta.gca)
+  	zData(id_remove) = [];
+  end
 
-      % Rather than calculating invA first and then multiply with rhs to obtain 
-      % lambda, directly calculate the respective terms
-      % The upper half of the 2x2 matrix is always the same and is given by:
-      % [-(X4(2)-X3(2)), X4(1)-X3(1)] / detA * rhs
-      % This is a matrix multiplication of the form [1x2] * [2x1] = [1x1]
-      % As we have transposed rhs we can write this as:
-      % rhs * Rotate * (X4-X3) => [nx2] * [2x2] * [2x1] = [nx1]
-      lambda(id_detA, 1) = (rhs(id_detA, :) * Rotate * (X4-X3))./detA(id_detA);
+  % Set the new data.
+  set(handle, 'XData', xData);
+  set(handle, 'YData', yData);
+  if isAxis3D(meta.gca)
+  	set(handle, 'zData', zData);
+  end
+end
+% =========================================================================
+function removeNaNs(meta, handle)
+  % Removes superflous NaNs in the data
```

Comment:
No this would remove *all* NaNs. However, we want only those at the beginning, the end and those that are doubled